tests/comparisons/Model/testBakeEnumBacked.php

```diff
@@ -0,0 +1,21 @@
+<?php
+declare(strict_types=1);
+
+namespace Bake\Test\App\Model\Enum;
+
+use Cake\Database\Type\EnumLabelInterface;
+use Cake\Utility\Inflector;
+
+/**
+ * FooBar Enum
+ */
+enum FooBar: string implements EnumLabelInterface
+{
+    /**
+     * @return string
+     */
+    public function label(): string
+    {
+        return Inflector::humanize(mb_strtolower($this->name));
```

Comment:
```suggestion
        return Inflector::humanize(Inflector::underscore($this->name));
```

This will ensure name like "FooBar" gets inflicted to "Foo Bar".